File: jdk/StudentProject_ver3/src/service/StudentService.java
package service;

import java.util.ArrayList;

import vo.StudentVO;

public class StudentService {
	private static StudentService instance = new StudentService();
	private ArrayList<StudentVO> list;

	private StudentService() {
		list = new ArrayList<StudentVO>();
	}
	
	public ArrayList<StudentVO> getList() {
		return list;
	}

	public static StudentService getInstance() {
		if (instance == null)
			instance = new StudentService();
		return instance;
	}

	public boolean insertStudent(StudentVO studentVO) {
		// 동일한 객체가 있는지 체크 -- 학번으로만 비교
		if (list.contains(studentVO))
			return false;
		list.add(studentVO);
		return true;
	}

	public StudentVO searchStudent(String studentNo) {
		StudentVO vo = new StudentVO(studentNo, null, null, 0);
		int no = list.indexOf(vo);
		if (no == -1)
			return null;
		return list.get(no);
	}

	public boolean deleteStudent(String studentNo) {
		StudentVO vo = new StudentVO(studentNo, null, null, 0);
		return list.remove(vo);
	}

}

File: C#/StudentProject/StudentProject/StudentService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StudentProject
{
    public class StudentService
    {
        private static readonly StudentService instance = new StudentService();
        private List<StudentVO> list;
        private StudentService()
        {
            list = new List<StudentVO>();
        }

        internal bool deleteStudent(string studentNo)
        {
            StudentVO studentVO = new StudentVO(studentNo, null, null, 0);
            return list.Remove(studentVO);
        }

        internal StudentVO searchStudent(string name)
        {
            foreach (StudentVO student in list)
            {
                if (student.Name.Contains(name))
                    return student;
            }
            return null;
        }

        public static StudentService Instance { get { return instance; } }

        public bool insertStudent(StudentVO studentVO)
        {
            if(list.Contains(studentVO)) return false;
            list.Add(studentVO);
            return true;
        }
    }
}
